subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -51,62 +42,133 @@ enum {
 	ADV_FLAGS_NUM
 };
 
-static struct {
+typedef struct net_buf* (*adv_buf_handler_t)(k_timeout_t timeout);
+
+struct ext_adv {
 	ATOMIC_DEFINE(flags, ADV_FLAGS_NUM);
 	struct bt_le_ext_adv *instance;
 	struct net_buf *buf;
 	uint64_t timestamp;
 	struct k_work_delayable work;
-} adv;
+	adv_buf_handler_t handler;
+	struct bt_le_adv_param adv_param;
+};
 
-static int adv_start(const struct bt_le_adv_param *param,
+static void send_pending_adv(struct k_work *work);
+static struct ext_adv adv_main = {
+	.work = Z_WORK_DELAYABLE_INITIALIZER(send_pending_adv),
+	.handler = bt_mesh_adv_buf_get,
+	.adv_param = {
+		.id = BT_ID_DEFAULT,
+		.interval_min = BT_MESH_ADV_SCAN_UNIT(ADV_INT_FAST_MS),
+		.interval_max = BT_MESH_ADV_SCAN_UNIT(ADV_INT_FAST_MS),
+#if defined(CONFIG_BT_MESH_DEBUG_USE_ID_ADDR)
+		.options = BT_LE_ADV_OPT_USE_IDENTITY,
+#endif
+	}
+};
+
+#if defined(CONFIG_BT_MESH_RELAY)
+#define RELAY_ADV_COUNT			CONFIG_BT_MESH_RELAY_ADV_SETS
```

Comment:
Since this is just a plain alias now, it'd be easier to follow if we just used `CONFIG_BT_MESH_RELAY_ADV_SETS` throughout